### PR TITLE
DATAES-217 Propose scan method with a class in order to retrieve inde…

### PR DESCRIPTION
**Pull request by kevinleturc**
…x and type from Document

### CODE REVIEW

File: src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchOperations.java [hunk 1]
```diff
@@ -491,6 +491,18 @@
...
 
 	/**
+	 * Returns scroll id for criteria query
+	 *
+	 * @param query
+	 * @param scrollTimeInMillis
+	 * @param noFields
+	 * @param clazz
+	 * @param <T>
+	 * @return
```
Comment (martin-g) [inline src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchOperations.java]: Javadoc is missing for all parameters and the return.

File: src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchOperations.java [hunk 2]
```diff
@@ -501,6 +513,18 @@
...
 
 	/**
+	 * Returns scroll id for scan query
+	 *
+	 * @param query
+	 * @param scrollTimeInMillis
+	 * @param noFields
+	 * @param clazz
+	 * @param <T>
+	 * @return
```
Comment (martin-g) [inline src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchOperations.java]: Javadoc missing.

File: src/test/java/org/springframework/data/elasticsearch/core/ElasticsearchTemplateTests.java [hunk 3]
```diff
@@ -902,6 +902,61 @@ public void shouldReturnResultsForScanAndScrollWithCustomResultMapperForGivenSea
 	DATAES-167
```
Comment (martin-g) [inline src/test/java/org/springframework/data/elasticsearch/core/ElasticsearchTemplateTests.java]: s/167/217/ ?

**Comment by kevinleturc** [comment]
I review according to your comments.

**Comment by kevinleturc** [comment]
Hi,
Any news on this ?
Regards

**Comment by kevinleturc** [comment]
Hi,
Any news on this ?
Regards

**Comment by akonczak** [comment]
Thanks @kevinleturc for the contribution.

**Pull request closed without merging**